CRM/Core/I18n.php

```diff
@@ -584,15 +584,14 @@ public function setGettextDomain($key) {
   }
 
   /**
-   * Is the CiviCRM in multilingual mode.
+   * Is the current CiviCRM domain in multilingual mode.
    *
    * @return Bool
    *   True if CiviCRM is in multilingual mode.
    */
   public static function isMultilingual() {
-    $domain = new CRM_Core_DAO_Domain();
-    $domain->find(TRUE);
-    return (bool) $domain->locales;
+    $domainId = CRM_Core_Config::domainID();
+    return (bool) CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Domain', $domainId, 'locales');
```

Comment:
Not a blocker/revertable thing -- but the function `CRM_Core_BAO_Domain::getDomain()` returns a cached instance of the active domain record, eg perhaps:

```php
return (bool) CRM_Core_BAO_Domain::getDomain()->locales;
```

I'm pretty sure `getDomain()` is used in a bunch of other places, so using it would avoid the extra query.

Reply:
@totten the results of `CRM_Core_DAO::getFieldValue` are cached in a static var.

I think the real problem with this PR is that it switches from reading domain 1 to reading the current domain, and it was an undocumented rule previously to only store locales data in domain 1. That's the part I want to revert but will have to think it through as the first record in the `civicrm_domain` table isn't guaranteed to have id `1`, so it's not as simple as changing `$domainId` to `1`.

Reply:
@eileenmcnaughton @seamuslee001 @totten we have multiple discussion threads about this and multiple open PRs.
I've changed my mind about the solution and I think having `locales` populated consistently for all domains is a good thing.
https://lab.civicrm.org/dev/core/-/issues/1852

Reply:
Aaaah, that makes sense.